packages/plugin-connector/src/data-sources/get-sources.js

```diff
@@ -0,0 +1,142 @@
+ * @param {string} sourcesDir The path to the sources directory
+ * @returns {Promise<DatasourceSpec[]>} An array of DatasourceSpecs
+ */
+export const getSources = async (sourcesDir) => {
+	const sourcesDirectories = await fs.readdir(sourcesDir);
+	/** @type {DatasourceSpec[]} */
+	const datasourceSpecs = await Promise.all(
+		sourcesDirectories.map(async (dirName) => {
+			const sourceDir = path.join(sourcesDir, dirName);
+			const contents = await fs.readdir(sourceDir);
+
+			const connParams = await getConnectionParams(sourceDir);
+			if (!connParams.name) connParams.name = /** @type {string} */ sourceDir.split(/[/\\]/).shift();
+			if (!connParams.name)
+				throw new Error(
+					`Unexpected error determining datasource name, please add an explicit name in connection.yaml (${sourceDir})`
+				);
+			// Load Options from Environment
+			connParams.options = { ...connParams.options, ...loadSourceOptions(connParams.name) };
+
+			const queries = await getQueries(sourceDir, contents);
+			return {
+				...connParams,
+				sourceDirectory: sourceDir,
+				queries: queries
+			};
+		})
+	).then((r) => r.filter(Boolean));
+
+	return datasourceSpecs;
+};
+
+/**
+ * Reads a YAML file containing connection parameters from the given source directory,
+ * parses it, and returns a validated datasource specification.
+ *
+ * @param {string} sourceDir - The directory containing the connection.yaml file.
+ * @return {Promise<DatasourceSpecFile>} A Promise that resolves to a validated datasource specification.
+ */
+async function getConnectionParams(sourceDir) {
+	const connParamsRaw = await fs
+		.readFile(path.join(sourceDir, 'connection.yaml'))
+		.then((r) => r.toString());
+
+	let connParamsUnchecked;
+	try {
+		connParamsUnchecked = yaml.parse(connParamsRaw);
+	} catch (e) {
+		throw new Error(`Error parsing connection.yaml file; ${sourceDir}`, { cause: e });
+	}
+
+	const validationResult = DatasourceSpecFileSchema.safeParse(connParamsUnchecked);
+	if (!validationResult.success) {
+		console.error(chalk.bold.red(`[!] connection.yaml has errors (${sourceDir}`));
+		const formattedError = cleanZodErrors(validationResult.error.format());
+		console.error(chalk.red('|   Discovered Errors:'));
+		const redPipe = chalk.red('|');
+		console.error(
+			`${redPipe}   ${yaml.stringify(formattedError).replace(/\n/g, `\n${redPipe}   `)}`
+		);
+		throw new Error('Unable to load connection.yaml');
+	}
+	return validationResult.data;
+}
+
+/**
+ * Retrieves the contents of all query files in the source directory,
+ * excluding the 'connection.yaml' file, and returns them as an array of
+ * objects containing the filepath and content of each query file.
+ *
+ * @param {string} sourceDir - The path to the source directory.
+ * @param {Array<string>} contents - An array of filenames in the source directory.
+ * @return {Promise<DatasourceQuery[]>} - A promise that resolves to an array of objects
+ * containing the filepath and content of each query file.
+ */
```

Comment:
Yes, it would be up to the duckdb plugin to filter that out. All files except the connection.yaml are passed to the plugin

We don't want to assume a specific extension for all queries, because non-sql adapters will probably have their own (e.g. .gql for graphql).